src/app/shared/models/domain/item-edit-permission.ts

```diff
@@ -0,0 +1,56 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import * as D from 'io-ts/Decoder';
+
+// short versions, not for export
+const
+  NONE = 'none',
+  CHILDREN = 'children',
+  ALL = 'all',
+  ALL_WITH_GRANT = 'all_with_grant';
+
+export const
+  ITEMEDITPERM_NONE = NONE,
+  ITEMEDITPERM_CHILDREN = CHILDREN,
+  ITEMEDITPERM_ALL = ALL,
+  ITEMEDITPERM_ALL_WITH_GRANT = ALL_WITH_GRANT,
+  ITEMEDITPERM_MAX = ALL_WITH_GRANT;
+
+export const itemEditPermValues = [ NONE, CHILDREN, ALL, ALL_WITH_GRANT ] as const;
+export const itemEditPermDecoder = D.struct({
+  canEdit: D.literal(...itemEditPermValues)
+});
+export type ItemEditPerm = D.TypeOf<typeof itemEditPermDecoder>;
+export interface ItemWithEditPerm { permissions: ItemEditPerm }
+
+/**
+ * Whether the permission allows the user/group to change the children (and the properties of the relation) of the item
+ */
+export function allowsEditingChildren(p: ItemEditPerm): boolean {
+  return [ CHILDREN, ALL, ALL_WITH_GRANT ].includes(p.canEdit);
+}
+/**
+ * Whether the permission allows the user/group to change children and all properties of the item (but cannot delete)
+ */
+export function allowsEditingAll(p: ItemEditPerm): boolean {
+  return [ ALL, ALL_WITH_GRANT ].includes(p.canEdit);
+}
+/**
+ * Whether the permission allows the user/group to give editing permissions to others (require perm on the receiver as well!)
+ */
+export function allowsGrantingEdition(p: ItemEditPerm): boolean {
+  return p.canEdit === ITEMEDITPERM_ALL_WITH_GRANT;
+}
+
+// ********************************************
+// Pipes for templates
+// ********************************************
+
+@Pipe({ name: 'allowsEditingChildren', pure: true })
+export class AllowsEditingChildrenItemPipe implements PipeTransform {
+  transform = allowsEditingChildren;
+}
+
+@Pipe({ name: 'allowsEditingAll', pure: true })
+export class AllowsEditingAllItemPipe implements PipeTransform {
+  transform = allowsEditingAll;
+}
```

Comment:
About pipes, since angular template system type-checking differs from the JS one and understands badly optional chaining, I think it would ease using those pipe if they tolerate nil value (and when nil they can return `false`)
That way we could use them as `*ngIf="itemData?.permissions | allowsEditingAll"`

Reply:
It was done on purpose, I did not want to accept falsy values as it ends up keeping bugs silenced.

Actually, I forgot to implement an extra commit that I wanted to implement last: add an assertion in these pipes, in dev throw an exception if falsy, and prod console.err if so.

What do you think about that?

Reply:
Not much, to me both approaches are valid, as long as we use one that works it's ok.